### PR TITLE
support generation of model from openapi schemas

### PR DESCRIPTION
**Pull request by MUzairS15**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — COMMENTED** [review]
@MUzairS15 build checks are failing...

**Comment by MUzairS15** [comment]
ensure correct version is passed.

**Review by leecalcote — COMMENTED** [review]
Still needed?

**Comment by MUzairS15** [comment]
// @shlok

File: utils/component/openapi_generator.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+				SchemaVersion: v1beta1.ModelSchemaVersion,
+				Version:       "v1.0.0",
+
+				Model: model.Model{
+					Version: pkg.GetVersion(),
+				},
+				Name:        pkg.GetName(),
+				DisplayName: manifests.FormatToReadableString(pkg.GetName()),
+				Category: category.CategoryDefinition{
+					Name: "Orchestration & Management",
```
Comment (MUzairS15) [inline utils/component/openapi_generator.go:124]: Update

File: utils/component/openapi_generator.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+			Component: component.Component{
+				Kind:    kind,
+				Version: apiVersion,
+				Schema:  string(crd),
+			},
+			// Metadata:    compMetadata,
+			DisplayName: manifests.FormatToReadableString(kind),
+			Model: model.ModelDefinition{
+				SchemaVersion: v1beta1.ModelSchemaVersion,
+				Version:       "v1.0.0",
```
Comment (leecalcote) [inline utils/component/openapi_generator.go:116]: Why is this still hardcoded?

File: utils/component/openapi_generator.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		DeleteFields(modified)
+		crd, err = json.Marshal(modified)
+		if err != nil {
+			fmt.Printf("%v", err)
+			continue
+		}
+
+		c := component.ComponentDefinition{
+			SchemaVersion: v1beta1.ComponentSchemaVersion,
+			Version:       "v1.0.0",
```
Comment (leecalcote) [inline utils/component/openapi_generator.go:104]: Hardcoding appropriate?

**Pull request merged**